content/docs/azimuth/layer2.md

```diff
@@ -0,0 +1,387 @@
+
+### Layer 2
+
+Layer 1 still functions identically today as it did before naive rollups. Naive
+rollups work via the following process.
+
+ 1. A batch of one or more transactions is posted to the Ethereum blockchain by
+    an urbit node called a roller or aggregator.
+ 2. Your urbit downloads the transactions from an Ethereum node.
+ 3. Your urbit computes the resulting state transitions from the transactions
+    and checks them for validity.
+ 4. Your urbit updates its locally stored Azimuth state using state transitions
+    from the batch that have been deemed valid.
+    
+ In comparison with Layer 1, the EVM no longer checks the validity or computes
+ the state transitions for any given transaction. It is now being used solely as
+ a database of submitted transactions, and the business logic of computing what
+ these transactions mean has been offloaded to your urbit. Thus we think of
+ `naive.hoon` as being the first "Hoon smart contract". You could also consider
+steps 3 and 4 of the Layer 2 process as being a fattening of the trivial step 4
+ of the layer 1 process.
+
+Here we briefly elaborate on the layer 2 steps, but see below for more technical
+detail.
+
+A roller is any urbit node - even a moon, comet, or fakezod will do - to which
+batches of transactions are submitted. You could use your own ship as a roller
+if you wanted. The roller collects batches of transactions from whichever
+parties they choose and submits them to the Ethereum blockchain all at once.
+We expect this to either happen on a regular interval, or once some minimum
+threshold of transactions is reached, but the decision on when to submit is
+ultimately up to the roller.
+
+Computing the resulting state transitions obtained from downloaded Ethereum
+transactions is done using `/lib/naive.hoon`, which is a gate that accepts both layer
+1 transaction logs and layer 2 batches, and then computes the resulting state
+transitions and updates the ship's internal Azimuth data held in
+the Gall agent `/app/azimuth.hoon` accordingly.
+
+### Savings in gas costs
+
+There are several dimensions by which naive rollups saves on gas over layer 1.
+They are:
+
+ 1. Gas is not spent on instructing the EVM to compute state transitions and
+    confirm validity.
+ 2. Layer 2 Azimuth state is not stored on Ethereum, so the only data storage
+    gas costs is for the transactions themselves.
+ 3. Layer 2 transactions are written in a highly compressed form. E.g., instead
+    of calling the spawn action by name, it is simply referred to as action `%1`.
+ 4. By collecting multiple layer 2 transactions and submitting them as a single
+    "batch transaction", additional gas savings are achieved by not needing to
+    duplicate information such as which smart contract the transactions are
+    intended for.
+    
+Put together, these create a reduction in gas costs of at least 65x. This is
+true even for batches of L2 transactions that contain only a single transaction.
```

Comment:
Actually that's for a marginal transaction (ie if you're already sending a batch, what does it cost to add another tx).  If it's a single transaction in a batch, you'll end up around 5x cheaper.  10 transactions is around 30x cheaper, 30 txs is about 65x cheaper.